CMake/FindTBB.cmake

```diff
@@ -1,10 +1,27 @@
-find_path(TBB_INCLUDE_DIR tbb/tbb.h)
```

Comment:
This is a direct copy of HOOMD's `FindTBB.cmake`. I wasn't able to make the previous version work.